src/sqlfluff/rules/L069.py

```diff
@@ -0,0 +1,175 @@
+"""Implementation of Rule L069."""
+
+from typing import Optional
+
+from sqlfluff.core.parser.segments.raw import CodeSegment
+from sqlfluff.core.rules.crawlers import SegmentSeekerCrawler
+from sqlfluff.core.rules import BaseRule, LintFix, LintResult, RuleContext
+from sqlfluff.utils.functional import sp, FunctionalContext
+from sqlfluff.core.rules.doc_decorators import (
+    document_configuration,
+    document_fix_compatible,
+    document_groups,
+)
+
+
+@document_groups
+@document_fix_compatible
+@document_configuration
+class Rule_L069(BaseRule):
+    """Use ``CAST`` instead of ``CONVERT`` or ``::``.
+
+    .. note::
+        This is only compatible with 2-arguments CONVERT as
+        some dialects allow an optional 3rd argument e.g TSQL,
+        which cannot be rewritten into CAST.
+        This rule is disabled by default for Teradata because it supports different
+        type casting apart from CONVERT and ::
+        e.g DATE '2007-01-01', '9999-12-31' (DATE).
+
+    **Anti-pattern**
+
+    Using CONVERT or :: instead of CAST when ``allow_non_standard_type_casting``
+    config is set to ``False`` (default).
+
+    .. code-block:: sql
+
+        SELECT convert(varchar, 1) AS bar
+        FROM foo;
+
+        SELECT 1::varchar AS bar
+        FROM foo;
+
+    **Best practice**
+
+    Use ``CAST`` instead.
+    ``CAST`` is universally supported
+
+    .. code-block:: sql
+
+        SELECT CAST(1 AS varchar) AS bar,
+        FROM foo;
+
+    """
+
+    groups = ("all",)
+    config_keywords = ["allow_non_standard_type_casting"]
+    crawl_behaviour = SegmentSeekerCrawler({"function", "cast_expression"})
+
+    def _eval(self, context: RuleContext) -> Optional[LintResult]:
+        """Use ``CAST`` instead of ``CONVERT`` or ``::``."""
+        # Config type hints
+        self.allow_non_standard_type_casting: bool
+
+        # Rule disabled for teradata.
+        if context.dialect.name == "teradata":
+            return None
+
+        if not self.allow_non_standard_type_casting:
```

Comment:
Don't this this True, false makes sense.

If False is the same as excluding the Rule in general.

PS this would be easier to see if it read

```python
if self.allow_non_standard_type_casting:
   return None
```

Reply:
Yeah, that works too.